resilience4j-bulkhead/src/main/java/io/github/resilience4j/bulkhead/adaptive/internal/config/AbstractConfig.java

```diff
@@ -0,0 +1,211 @@
+/*
+ *
+ *  Copyright 2019: Mahmoud Romeh
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package io.github.resilience4j.bulkhead.adaptive.internal.config;
+
+import java.time.Duration;
+import java.util.Objects;
+
+import io.github.resilience4j.bulkhead.BulkheadConfig;
+import io.github.resilience4j.core.lang.NonNull;
+
+/**
+ * abstract common configuration
+ */
+public class AbstractConfig {
+	private static final SlidingWindow DEFAULT_SLIDING_WINDOW_TYPE = SlidingWindow.COUNT_BASED;
+	private static final double DEFAULT_FAILURE_RATE_THRESHOLD = 50.0; // Percentage
+	private static final double DEFAULT_SLOW_CALL_RATE_THRESHOLD = 50.0; // Percentage
+	private static final int DEFAULT_SLIDING_WINDOW_SIZE = 100;
+	private static final long DEFAULT_SLOW_CALL_DURATION_THRESHOLD = 5; // Seconds
+	private static final int DEFAULT_SLIDING_WIN_TIME = 10; // Seconds
+
+	double failureRateThreshold = DEFAULT_FAILURE_RATE_THRESHOLD;
+	int slidingWindowSize = DEFAULT_SLIDING_WINDOW_SIZE;
+	SlidingWindow slidingWindowType = DEFAULT_SLIDING_WINDOW_TYPE;
+	double slowCallRateThreshold = DEFAULT_SLOW_CALL_RATE_THRESHOLD;
+	Duration desirableLatency = Duration.ofSeconds(DEFAULT_SLOW_CALL_DURATION_THRESHOLD);
```

Comment:
The name `desirableLatency` is nice. But maybe we should use the same name like the CircuitBreakerConfig does which is `slowCallDurationThreshold`